internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -644,3 +648,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	xr.SetConditions(xpv1.Available())
 	return reconcile.Result{RequeueAfter: r.pollInterval}, errors.Wrap(r.client.Status().Update(ctx, xr), errUpdateStatus)
 }
+
+func firstNAndSomeMore(names []string) string {
```

Comment:
would it make sense to sort names first? just to avoid flapping, which should not happen here atm, but could happen with future changes.

Reply:
they are sorted by the order in the compisition.

Reply:
We could order them alphabetically. That's merely cosmetics. No strong opinion.

Reply:
yes, I meant alphabetically, and yes, that's why I was saying it wouldn't change much here at the moment, but it could change in the future if we changed the logic for any reason. However, no strong opinion either.

Reply:
Personally I would prefer the order from the Composition.  I tend to write my Compositions with the resources in the order that I would expect them to be created, which makes internal dependencies easier to follow.  So having the not-ready resources listed in the order they show up in the Composition would also serve to indicate "how far along" the Composite is in the creation process, and give an indication (to the author) of what might be hanging up.

Granted that relies on the resources being specified in a certain order, but I don't think there is any inherent value in alphabetizing them that would make the debugging process any easier.

Either way having the information is a huge help.

Reply:
We won't be able to guarantee a stable order with Composition Functions, so we'll need to defensively sort somehow.

Reply:
Added the sorting.